### PR TITLE
trying to upgrade to ts 4.9.5

### PR DESCRIPTION
**Pull request by foxriver76**
@AlCalzone need you here 
batsignal

### CODE REVIEW

**Comment by AlCalzone** [comment]
The only thing left I see is that comment where it infers `state | design` for you but not for me. Or is that solved now?

**Comment by foxriver76** [comment]
ah no this is no longer true, I have removed the comments

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 1]
```diff
@@ -2703,19 +2700,16 @@ export class AdapterClass extends EventEmitter {
...
+                if ('custom' in options.obj.common && options.obj.common.custom !== null) {
                     this._logger.warn(
                         `${this.namespaceLog} setObject ${options.id} (type=${options.obj.type}) property common.custom must not exist.`
                     );
                     delete options.obj.common.custom;
                 }
             }
 
-            if (!Object.prototype.hasOwnProperty.call(options.obj.common, 'name')) {
+            if (options.obj.common && !Object.prototype.hasOwnProperty.call(options.obj.common, 'name')) {
```
Comment (AlCalzone) [inline packages/adapter/src/lib/adapter/adapter.ts:2712]: Maybe
```suggestion
            if (options.obj.common && !('name' in options.obj.common)) {
```

Comment (AlCalzone) [reply packages/adapter/src/lib/adapter/adapter.ts:2712]: Unless that breaks something.

**Review by AlCalzone — COMMENTED** [review]
Noticed one thing, everything else looks good

File: packages/adapter/src/lib/adapter/adapter.ts (hunk 1 above)
Comment (foxriver76) [reply packages/adapter/src/lib/adapter/adapter.ts:2712]: I tried this, but then in the assignment below he says `name` does not exist on never, has common "always" has name

**Pull request merged**